README.md

```diff
@@ -1,2 +1,171 @@
+Portal for learners and course teams to access MITx Micromasters programs
+
+## Getting Started
+
+Although you can run Micromasters locally with a default sqlite database after
+installing the ``requirements.txt`` file, the recommended way is to 
+use [Docker](https://www.docker.io). Install Docker, then install 
+``docker-compose`` and run the app: 
+
+    pip install docker-compose 
+    docker-compose up
+
+### (OSX only) Create your docker machine:
+
+    docker-machine create default
+    docker-machine start default
+    docker-machine env default
+    eval "$(docker-machine env default)"
+
+These commands create a Docker container named ``default``, start the
+container, and configure environment variables to facilitate communication
+with the ``edX`` instance.
+
+
+This will set up a near production-ready containerized development 
+environment that runs migrations, with the Django development server 
+running on port ``8079``.
+
+To run one-off commands, like shell, you can run:
+
+    docker-compose run web python manage.py shell
+
+or to create root user:
+
+    docker-compose run web python manage.py createsuperuser
+
+
+## For OS X Development
+
+Due to issues using Docker in OSX, development is more efficient if you
+install [docker-osx-dev](https://github.com/brikis98/docker-osx-dev).
+docker-osx-dev synchronizes host file-system changes with the Docker
+container. If you have [``homebrew``](http://brew.sh/) installed, you can 
+install ``docker-osx-dev`` by typing ``make`` in the root of the Micromasters
+project directory. The make file will install or update Docker using Homebrew
+and go on to install ``docker-osx-dev``.
+
+Subsequently, before you run the application with ``docker-compose up``, 
+you would run:
+  
+    docker-osx-dev -m default -s ./ --ignore-file '.rsync-ignore'
+
+(Assuming your Docker VM is called ``default``, and your current working
+directory is the root of the ``micromasters`` source directory).
+
+This starts a process that monitors the file system for changes. On startup
+you may receive this error:
+  
+      [ERROR] Found VirtualBox shared folders on your Boot2Docker VM. 
+      These may void any performance benefits from using docker-osx-dev:
+      
+      /Users
+      
+      [INSTRUCTIONS] Would you like this script to remove them?
+      1) yes
+      2) no
+
+Answer ``yes``.
+
+## Configuration and Start-up
+
+Create a user who has permission to view the course:
+
+    docker-compose run web python manage.py createsuperuser
+
+Start the machine:
+
+    docker-compose up
+
+(OSX only) visit: http://192.168.99.100:8079/
+
+  Note: Your IP address may vary depending on what address docker 
+  assigns to your VM.  If the IP address above doesn't work:
+
+  1. Power down your docker-machine (``docker-machine stop default``)
+  2. ``docker-machine ssh default``
+  3. Once in the docker machine shell, run ``ifconfig``, and look for
+     the IP address that begins with ``192.168.99.xxx``.
```

Comment:
there is a quicker way to get the IP that is
`docker-machine ls`

this will give an output like
```
NAME             ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER    ERRORS
ccxcon           -        virtualbox   Stopped                                       Unknown
dev              -        virtualbox   Stopped                                       Unknown
lore             -        virtualbox   Stopped                                       Unknown
micromasters     -        virtualbox   Running   tcp://192.168.99.100:2376           v1.10.2
teachersportal   -        virtualbox   Stopped                                       Unknown
```